File: cpp/41e_1097A_GennadyAndACardGame.cpp
#include<bits/stdc++.h>

int main(){
    std::string tab_c, c1, c2, c3, c4, c5;
    std::cin>>tab_c;
    std::cin>>c1>>c2>>c3>>c4>>c5;
    char chck_r = tab_c[0], chck_s = tab_c[1];
    if(c1[0] == chck_r || c2[0] == chck_r || c3[0] == chck_r || c4[0] == chck_r || c5[0] == chck_r){
        std::cout<<"YES";
    }
    else if(c1[1] == chck_s || c2[1] == chck_s || c3[1] == chck_s || c4[1] == chck_s || c5[1] == chck_s){
        std::cout<<"YES";
    }
    else{
        std::cout<<"NO";
    }
    return 0;
}

File: python/41e_1097A_GennadyAndACardGame.py
def main():
    t_c = input().strip()
    c1, c2, c3, c4, c5 = input().split(" ")
    chck_r, chck_s = t_c[0], t_c[1]
    if c1[0] == chck_r or c2[0] == chck_r or c3[0] == chck_r or c4[0] == chck_r or c5[0] == chck_r:
        print("YES")
    elif c1[1] == chck_s or c2[1] == chck_s or c3[1] == chck_s or c4[1] == chck_s or c5[1] == chck_s:
        print("YES")
    else:
        print("NO")

if __name__ == "__main__":
    main()